src/libraries/System.Linq/src/System/Linq/Match.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+
+namespace System.Linq
+{
+    public static partial class Enumerable
+    {
+        public static (List<TSource> Matched, List<TSource> Unmatched) Match<TSource>(this IEnumerable<TSource> source, Func<TSource, bool> predicate)
```

Comment:
For new APIs, we use XML comments to seed the documentation (and the src is also becoming the source of truth for docs moving forward).  This would need an XML comment on the method describing it, the parameters, the return type, exceptions, etc.